src/Modbus.Common/Util/Extensions.cs

```diff
@@ -85,7 +86,11 @@ public static uint GetUInt32(this IEnumerable<ModbusObject> list, int startIndex
 			if (!list.All(r => r.Type == ModbusObjectType.HoldingRegister) && !list.All(r => r.Type == ModbusObjectType.InputRegister))
 				throw new ArgumentException("Invalid register type");
 
-			var registers = list.Skip(startIndex).Take(2).ToArray();
+			list = list.OrderBy(r => r.Address).Skip(startIndex).Take(2);
+			if (!bigEndianRegistryOrder)

```

Comment:
```suggestion
			if (inverseRegisters)

```

... so it's easier to understand that an action takes place when a parameter is actually set.